build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/MrjarPlugin.java

```diff
@@ -163,6 +163,7 @@ private void addJar(Project project, SourceSet sourceSet, int javaVersion) {
         project.getConfigurations().register("java" + javaVersion);
         TaskProvider<Jar> jarTask = project.getTasks().register("java" + javaVersion + "Jar", Jar.class, task -> {
             task.from(sourceSet.getOutput());
+            task.getArchiveClassifier().set("java" + javaVersion);
```

Comment:
I don't know enough about how this works... will this select the newest applicable one, and default to `java` if there are none?

Reply:
No, this is about the output artifact name. Previously these different tasks for eg java 21 and 23 would create the same jar name. By adding a classifier, each is unique and won't overwrite each other.